src/components/Planning/ProjectProgrammedDialog/ProjectProgrammedDialog.tsx

```diff
@@ -0,0 +1,176 @@
+    if (value) {
+      setFormState((current) => ({
+        ...current,
+        projectsForSubmit: [...current.projectsForSubmit, value],
+      }));
+    }
+  }, []);
+
+  const onProjectSelectionDelete = useCallback((projectName: string) => {
+    setFormState((current) => ({
+      ...current,
+      projectsForSubmit: current.projectsForSubmit.filter((p) => {
+        return p.label !== projectName;
+      }),
+    }));
+  }, []);
+  const { projectsForSubmit } = formState;
+
+  const { Header, Content, ActionButtons } = Dialog;
+
+  const handleDialogClose = useCallback(() => {
+    setFormState((current) => ({ ...current, projectsForSubmit: [] }));
+    handleClose();
+  }, [handleClose]);
+
+  const onSubmit = useCallback(
+    async (e: MouseEvent<HTMLButtonElement>) => {
+      e.preventDefault();
+      dispatch(silentPatchMultipleProjectsThunk(buildRequestPayload(projectsForSubmit))).then(
+        () => {
+          setFormState((current) => ({
+            ...current,
+            projectsForSubmit: [],
+          }));
+        },
+      );
+    },
+
+    [dispatch, buildRequestPayload, projectsForSubmit],
+  );
+
+  return (
+    <div>
+      {/* Dialog */}
+      <div className="display-flex-col">
+        <Dialog
+          id="add-project-programmed-dialog"
+          aria-labelledby={'add-project-programmed-dialog-label'}
+          isOpen={isOpen}
+          close={handleDialogClose}
+          closeButtonLabelText={t('closeGroupFormWindow')}
+          className="big-dialog"
+          scrollable
+        >
+          {/* Header */}
+          <Header id={'add-project-programmed-header'} title={t(`addProjectsToProgramming`)} />
+
+          <Content>
+            <div className="dialog-search-section">
+              <br />
+
+              <div>
+                <ProjectProgrammedSearch
+                  onProjectClick={onProjectClick}
+                  onProjectSelectionDelete={onProjectSelectionDelete}
+                  projectsForSubmit={projectsForSubmit}
+                />
+              </div>
+            </div>
+          </Content>
+          <ActionButtons>
+            <Button
+              onClick={onSubmit}
+              disabled={!(projectsForSubmit && projectsForSubmit.length > 0)}
+              data-testid="search-projects-button"
+            >
+              {t('search')}
+            </Button>
+            <Button onClick={handleDialogClose} variant="secondary" data-testid="cancel-search">
+              {t('cancel')}
+            </Button>
+          </ActionButtons>
+        </Dialog>
+      </div>
+    </div>
+  );
+};
+
+const ProjectProgrammedDialog: FC = () => {
+  const selectedClass = useAppSelector(selectSelectedClass);
+  const selectedSubClass = useAppSelector(selectSelectedSubClass);
+  const { t } = useTranslation();
+  const [DialogButtonState, setDialogButtonState] = useState<IDialogButtonState>({
```

Comment:
As a note, variables shouldn't be named with uppercase letters. That should be used for types, interfaces, enums, classes and components. However, this state could just be converted to:

`const [isOpen, setIsOpen] = useState(false);`